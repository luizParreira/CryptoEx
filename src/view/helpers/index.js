// @flow

// Generate random string to be used as key to the loops
export const generateKey = (n: number = 36): string => Math.random().toString(n);
