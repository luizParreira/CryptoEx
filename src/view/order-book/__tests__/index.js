import React from 'react';
import renderer from 'react-test-renderer';
import {orders} from './test.helpers';
import {OrderBook} from '..';

describe('OrderBook', () => {
  const isOrdersLoading = false;
  const props = {orders, isOrdersLoading};

  test('renders successfully with orders', () => {
    const expectedComponent = renderer.create(<OrderBook {...props} />);

    expect(expectedComponent).toMatchSnapshot();
  });

  test('renders successfully with empty orders', () => {
    const expectedComponent = renderer.create(<OrderBook orders={[]} isOrdersLoading={false} />);

    expect(expectedComponent).toMatchSnapshot();
  });

  test('renders successfully when loading', () => {
    const expectedComponent = renderer.create(<OrderBook isOrdersLoading orders={orders} />);

    expect(expectedComponent).toMatchSnapshot();
  });
});
