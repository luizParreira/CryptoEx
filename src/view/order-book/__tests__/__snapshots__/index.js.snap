// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderBook renders successfully when loading 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "marginBottom": 0,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 0,
      },
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#c6c4c4",
            "borderBottomWidth": 0.5,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginLeft": 5,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#c6c4c4",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "left",
              },
            ]
          }
        >
          Buy
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginLeft": 5,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#c6c4c4",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "left",
              },
            ]
          }
        >
          Sell
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "minHeight": 100,
          },
        ]
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`OrderBook renders successfully with empty orders 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "marginBottom": 0,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 0,
      },
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#c6c4c4",
            "borderBottomWidth": 0.5,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginLeft": 5,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#c6c4c4",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "left",
              },
            ]
          }
        >
          Buy
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginLeft": 5,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#c6c4c4",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "left",
              },
            ]
          }
        >
          Sell
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`OrderBook renders successfully with orders 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "marginBottom": 0,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 0,
      },
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#c6c4c4",
            "borderBottomWidth": 0.5,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginLeft": 5,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#c6c4c4",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "left",
              },
            ]
          }
        >
          Buy
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginLeft": 5,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#c6c4c4",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "left",
              },
            ]
          }
        >
          Sell
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 44,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginLeft": 5,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 18,
                },
              ]
            }
          >
            6
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          color="green"
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "green",
                "fontSize": 18,
              },
            ]
          }
        >
          $ 135.0
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 44,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginLeft": 5,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            color="red"
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "red",
                  "fontSize": 18,
                },
              ]
            }
          >
            $ 129.0
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 18,
              },
            ]
          }
        >
          5
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 44,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginLeft": 5,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 18,
                },
              ]
            }
          >
            9
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          color="green"
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "green",
                "fontSize": 18,
              },
            ]
          }
        >
          $ 228.0
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 44,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginLeft": 5,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            color="red"
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "red",
                  "fontSize": 18,
                },
              ]
            }
          >
            $ 133.0
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 18,
              },
            ]
          }
        >
          16
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 44,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginLeft": 5,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 18,
                },
              ]
            }
          >
            10
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          color="green"
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "green",
                "fontSize": 18,
              },
            ]
          }
        >
          $ 200.0
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 44,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginLeft": 5,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            color="red"
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "red",
                  "fontSize": 18,
                },
              ]
            }
          >
            $ 222.0
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 18,
              },
            ]
          }
        >
          15
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
